File: app/src/main/java/com/tryden/leetcode/java/easy/RomanToInt.java
package com.tryden.leetcode.java.easy;


/*
 *
 * [13] Roman to Integer
 *
 * https://leetcode.com/problems/roman-to-integer/
 *
 */

public class RomanToInt {
    public static void main(String[] args) {
        String s = "LVII";
        romanToInt(s);
    }

    public static int romanToInt(String s) {
        int sum = 0;
        int len = s.length();
        for (int i = 0; i < len; i++) {
            if (i < len-1 && getValue(s.charAt(i)) < getValue(s.charAt(i+1))) {
                sum -= getValue(s.charAt(i));
            } else {
                sum += getValue(s.charAt(i));
            }
        }
        return sum;
    }

    public static int getValue(char ch){
        switch (ch) {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
        }
        throw new Error("Illegal");
    }
}

File: app/src/main/java/com/tryden/leetcode/kotlin/easy/RomanToInt.kt
package com.tryden12.leetcode_kotlin.easy

/*
 *
 * [13] Roman to Integer
 *
 * https://leetcode.com/problems/roman-to-integer/
 *
 */

    fun main() {

    }

    fun romanToInt(s: String): Int {
        val map = mutableMapOf(
            'I' to 1, 'V' to 5, 'X' to 10, 'L' to 50, 'C' to 100, 'D' to 500, 'M' to 1000
        )
        var number = 0
        var last = 1000
        s.forEach {
            val value = map[it] ?: 0
            println(value)
            if (value > last) number -= last * 2
            println("value: "+ value + ", last: " + last)
            number += value
            last = value
        }
        return number
    }